app/pages/_app.tsx

```diff
@@ -51,16 +51,25 @@ export default class App extends NextApp<AppProps> {
       })
     );
 
+    // Safely parse from URL query string, returns {} if query string is invalid JSON
+    const safeParse = (toParse) => {
+      try {
+        return JSON.parse(toParse as string);
+      } catch (e) {
+        return {};
+      }
+    };
+
```

Comment:
This function probably belongs to `app/lib` to make it reusable and independently testable